### ISSUE TITLE
Azure Functions work Locally, but Timeout or Freeze when deployed on Azure

### ISSUE THREAD

**Issue by ruyakubu** [Azure/azure-functions-java-worker#140]
Has anyone been able to successfully run deployed Java Azure Functions in Azure?

I find it odd that none of these published samples work Locally, but all fail and keep Timing out on the Azure portal:
•	https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-first-java-maven
•	https://github.com/Microsoft/function-demo-java-on-azure

What's missing?

**Comment by pragnagopa** [comment]
can you please share function app name and approximate time when you saw timeouts?

**Comment by brunoborges** [comment]
Look at https://github.com/Azure/azure-functions-java-library/issues/16

**Comment by pragnagopa** [comment]
the sample is using old version of library which is not supported any more https://github.com/Microsoft/function-demo-java-on-azure/blob/master/walkthrough/pom.xml#L24
Will submit a PR to fix the sample.

**Comment by ruyakubu** [comment]
Which version works?

I am using version 1.10.

1) Here's my repo:
https://github.com/ruyakubu/helloJavaAzureFunction

2) Here's my maven archetype:
mvn archetype:generate -DarchetypeVersion=1.10 ^
    -DarchetypeGroupId=com.microsoft.azure ^
    -DarchetypeArtifactId=azure-functions-archetype

3) Here's the deployment azure link to show the Timeouts:
https://hello-function-20180711170135136.azurewebsites.net/api/hello?name=HelloWorld

**Comment by pragnagopa** [comment]
You are using old sdk which is not supported any more
```
<dependency>
            <groupId>com.microsoft.azure</groupId>
            <artifactId>azure-functions-java-core</artifactId>
            <version>1.0.0-beta-3</version>
</dependency>
```

Latest maven archetype should work. You do not have to specify a version.

cc: @jdneo

**Comment by ruyakubu** [comment]
what about the archetypeVersion=1.10?  Is that incorrect too?

**Comment by pragnagopa** [comment]
Yes. Just use the latest using command
```
mvn archetype:generate ^
    -DarchetypeGroupId=com.microsoft.azure ^
    -DarchetypeArtifactId=azure-functions-archetype
```

**Comment by ruyakubu** [comment]
Using the maven archetype without specifying the version gives my the following error when I try to deploy to Azure:

__[INFO] Scanning for projects...
[INFO]
[INFO] -----------------< com.test.functions:test-functions >------------------
[INFO] Building Azure Java Functions 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- azure-functions-maven-plugin:1.0.0-beta-2:deploy (default-cli) @ test-functions ---
AI: INFO 12-07-2018 09:09, 1: Configuration file has been successfully found as resource
AI: INFO 12-07-2018 09:09, 1: Configuration file has been successfully found as resource
[INFO] Starting deploying to Azure Functions test-functions-20180712090403798...
[INFO] Authenticate with Azure CLI 2.0
[INFO] Updating Azure Functions...
[INFO] Successfully updated Azure Functions test-functions-20180712090403798
[INFO]
[INFO] Step 1 of 4: Creating ZIP package...
[INFO] Successfully saved ZIP package at C:\mypath\test-functions\target\azure-functions\test-functions-20180712090403798.zip
[ERROR] Application setting 'AzureWebJobsStorage' not found.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 32.656 s
[INFO] Finished at: 2018-07-12T09:09:45-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.microsoft.azure:azure-functions-maven-plugin:1.0.0-beta-2:deploy (default-cli) on project test-functions: Application setting 'AzureWebJobsStorage' not found. -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException__

**Comment by pragnagopa** [comment]
@ruyakubu - Your sample is still using old sdk. Please see here for the updated sample https://github.com/pragnagopa/function-demo-java-on-azure

@jdneo - Any ideas on deploy error?

**Comment by ruyakubu** [comment]
Can you list step by step what steps a developer is supposed to follow?

1)  You advised me use an archetype w/o the version that generates the java Function + POM.xml.  I tried that and the azure deploy fails to deploy and has ERRORS (shown above)

2) At what point do we use the maven archetype; and at what point do we manually override everything that the archetype automatically generates?

3) Before I go on another wild goose hunt, can you git my Hello repo in your local environment and get it to successfully DEPLOY and RUN in the cloud, before you advised me to use another setting that does not work?   I think it makes more sense for you to use my exact code and replicate the error before you diagnose suggest another solution. 

4) Going in circles in this thread is not helping.  Start w/ my broken code, then point out a true solution

**Comment by pragnagopa** [comment]
FYI @brunoborges @Selvasingh @asavaritayal 
I submitted a PR with the fix to your sample code https://github.com/ruyakubu/helloJavaAzureFunction/pull/1

I was able to repro the deploy issue. Root cause is resource group name is not unique. Opened issue
https://github.com/Microsoft/azure-maven-plugins/issues/242

**Comment by jdneo** [comment]
@ruyakubu 

I see an error goes: `[ERROR] Application setting 'AzureWebJobsStorage' not found.`

Could you please check the Application settings of the Function App that `AzureWebJobsStorage` is not empty?

you can find this in Azure Portal:
![image](https://user-images.githubusercontent.com/6193897/42669608-ead98670-8688-11e8-9a48-62667006a53d.png)

**Comment by pragnagopa** [comment]
Closing this as this is fixed.
@jdneo - Please see Microsoft/azure-maven-plugins#242 for root cause.

**Comment by wkarratoon** [comment]
Kind of more related to [jdneo's comment](https://github.com/Azure/azure-functions-java-worker/issues/140#issuecomment-404708243) above but empty settings can also occur when you use --force on deploy to switch the FUNCTIONS_WORKER_RUNTIME of your function app, in my case to node. When I did so I first got a 503 waiting for a response. 

After some head-banging, I was then clued by comments here for this issue to check my own app settings and indeed found no value was selected for Node.js Version in Stack Settings. Picking a version resolved the issue. 

So use --force with caution. Microsoft could also enhance documentation guidance here and set default elections.
